src/world/action/land.c

```diff
@@ -1,9 +1,206 @@
 #include "common.h"
+#include "functions.h"
 
-INCLUDE_ASM(void, "world/action/land", func_802B6000_E24920, void);
+void func_802B62CC_E24BEC(void);
+void func_802B644C_E24D6C(void);
 
-INCLUDE_ASM(void, "world/action/land", func_802B61C0_E24AE0, void);
+void func_802B6000_E24920(void) {
+    f32 inputMoveMagnitude;
+    f32 inputMoveAngle;
+    s32 jumpInputCheck;
+    s32 phi_a0;
+    CollisionStatus* currentCollisionStatus = &gCollisionStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* currentCameras = &gCameras[0];
+    
+    if ((playerStatus->animFlags & 0x1000) != 0) {
+        func_802B62CC_E24BEC();
+        return;
+    }
 
-INCLUDE_ASM(void, "world/action/land", func_802B62CC_E24BEC, void);
+    if (playerStatus->flags < 0) {
+        playerStatus->flags = playerStatus->flags & 0x7F77FFF1;
+        playerStatus->fallState = 0;
+        playerStatus->decorationList = 0;
+        playerStatus->unk_C2 = 0;
+        playerStatus->decorationPos[0] = playerStatus->position.x;
+        playerStatus->decorationPos[1] = playerStatus->position.z;
 
-INCLUDE_ASM(void, "world/action/land", func_802B644C_E24D6C, void);
+        if ((playerStatus->animFlags & 0x4000) != 0) {
+            phi_a0 = 0x9 << 16;
+            phi_a0 |= 0x2;
+
+        } else {
+            phi_a0 = 0x6000B;
+            if ((playerStatus->animFlags & 1) == 0) {
+                phi_a0 = 0x1 << 16;
+                phi_a0 |= 0x9;
+            }
+        }
+        
+        suggest_player_anim_clearUnkFlag(phi_a0);
+        sfx_play_sound_at_player(0x8161, 0);
+        sfx_play_sound_at_player(0x148, 0);
+
+        if (((u16) currentCollisionStatus->currentFloor & 0x4000) == 0) {
+            phys_adjust_cam_on_landing();
+        }
+
+        currentCollisionStatus->lastTouchedFloor = -1;
+        playerStatus->animFlags &= 0xFFFB0000 | 0xFFFF;
```

Comment:
This is actually just the constant 0xFFFBFFFF, and corresponds to a single flag that's inverted, preferably, you'd find the corresponding anim flag and do ~WHATEVER_FLAG